CODING_STYLE.md

```diff
@@ -0,0 +1,98 @@
+Coding Standards
+================
+
+The coding standards for volatility are mostly by our linter and our code formatter.
+All code submissions will be vetted automatically through tests from both and the submission will not be accepted if either of these fail.
+
+Code Linter: Ruff
+Code Formatter: Black
+
+In additiom, there are some coding practices that we employ to prevent specific failure cases and ensure consistency across the codebase.  These are documented below along with the rationale for the decision.
+
+This is heavily based upon https://google.github.io/styleguide/pyguide.html with minor modifications for volatility use.
+
+Imports
+-------
+
+Use import statements for packages and modules only, not for individual types, classes, or functions and ideally not aliased naming would cause confusion.  This is to prevent people importing an imported method (which can lead to confusion and add in an unnecessary dependency in the import chain).
+
+* Use `import x` for importing packages and modules.
+* Use `from x import y where x` is the package prefix and y is the module name with no prefix.
+* Use `from x import y as z` in any of the following circumstances:
+    * Two modules named `y` are to be imported.
+    * `y` conflicts with a top-level name defined in the current module.
+    * `y` conflicts with a common parameter name that is part of the public API (e.g., `features`).
+    * `y` is an inconveniently long name.
+    * `y` is too generic in the context of your code (e.g., `from storage.file_system import options as fs_options`).
+
+Exemptions from this rule:
+
+    * Symbols from the following modules are used to support static analysis and type checking:
+        * `typing` module
+        * `collections.abc` module
+        * `typing_extensions` module
+
+Function calls
+--------------
+
+For longer function calls, where line length is no longer an issue, favour using keyword arguments for clarity over unnamed positional arguments.
```

Comment:
```suggestion
For longer function calls, where line length is no longer an issue, favor using keyword arguments for clarity over unnamed positional arguments.
```

Reply:
That's the UK spelling so until we decide on a region for documentation, I'm going to continue writing it natively...